cpp/include/cudf/detail/sequence.hpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/filling.hpp>
+#include <cudf/detail/sequence.hpp>
+#include <cudf/types.hpp>
+
+namespace cudf {
+namespace experimental {
+namespace detail {
+
+/**
+ * @copydoc cudf::experimental::sequence(size_type size, scalar const& init, scalar const& step,
+ *                                       rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource())
+ *
+ * @param stream CUDA stream to run this function
+ **/
+std::unique_ptr<column> sequence(size_type size, scalar const& init, scalar const& step,
+                                 rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                                 cudaStream_t stream = 0);
+
+/**
+ * @copydoc cudf::experimental::sequence(size_type size, scalar const& init,
+                                         rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource())
+ *
+ * @param stream CUDA stream to run this function
+ **/
+std::unique_ptr<column> sequence(size_type size, scalar const& init,
```

Comment:
Hmmm, so this API only accepts the init from device memory. What if the value is only known in host memory? Then a H2D copy is necessary, when it could be implemented (at the low level) with passing a parameter to the kernel, e.g. using a counting iterator passed to thrust.  Someday this might come up and we will need another version that takes a host-side scalar value.

Reply:
> Hmmm, so this API only accepts the init from device memory. What if the value is only known in host memory?

I think this is more of a flaw with `cudf::scalar` than this particular function.